src/metabase/models/secret.clj

```diff
@@ -25,7 +30,34 @@
           :can-write?        i/superuser?}))
 
 ;;; ---------------------------------------------- Hydration / Util Fns ----------------------------------------------
-;; none yet
+
+(defn value->string
+  "Returns the value of the given `secret` instance as a String."
+  {:added "0.42.0"}
+  ^String [{:keys [^bytes value] :as secret}]
+  (String. value StandardCharsets/UTF_8))
+
+(defn value->file!
+  "Returns the value of the given `secret` instance in the form of a file."
+  {:added "0.42.0"}
+  [{:keys [id ^bytes value] :as secret}]
+  (if (= :file-path (:source secret))
+    (let [secret-val          (value->string secret)
+          ^File existing-file (File. secret-val)]
+      (if (.exists existing-file)
```

Comment:
should this be a when? It just returns the existing file for no purpose. Seems its just `(when-not condition throw)`